pkgs/tools/networking/v2ray/default.nix

```diff
@@ -1,69 +1,63 @@
-{ lib, fetchFromGitHub, fetchurl, symlinkJoin, buildGoModule, runCommand, makeWrapper, nixosTests
-, v2ray-geoip, v2ray-domain-list-community, assets ? [ v2ray-geoip v2ray-domain-list-community ]
+{ lib, fetchFromGitHub, symlinkJoin, buildGoModule, makeWrapper, nixosTests
+, nix-update-script
+, v2ray-geoip, v2ray-domain-list-community
+, assets ? [ v2ray-geoip v2ray-domain-list-community ]
 }:
 
-let
-  version = "4.45.0";
+buildGoModule rec {
+  pname = "v2ray-core";
+  version = "5.1.0";
 
   src = fetchFromGitHub {
     owner = "v2fly";
     repo = "v2ray-core";
     rev = "v${version}";
-    sha256 = "sha256-vVCWCppGeAc7dwY0fX+G0CU3Vy6OBPpDBUOBK3ykg60=";
+    hash = "sha256-87BtyaJN6qbinZQ+6MAwaK62YzbVnncj4qnEErG5tfA=";
   };
 
-  vendorSha256 = "sha256-TbWMbIT578I8xbNsKgBeSP4MewuEKpfh62ZbJIeHgDs=";
+  # `nix-update` doesn't support `vendorHash` yet.
+  # https://github.com/Mic92/nix-update/pull/95
+  vendorSha256 = "sha256-RuDCAgTzqwe5fUwa9ce2wRx4FPT8siRLbP7mU8/jg/Y=";
 
-  assetsDrv = symlinkJoin {
-    name = "v2ray-assets";
-    paths = assets;
-  };
+  ldflags = [ "-s" "-w" "-buildid=" ];
 
-  core = buildGoModule rec {
-    pname = "v2ray-core";
-    inherit version src;
+  subPackages = [ "main" ];
 
-    inherit vendorSha256;
-
-    doCheck = false;
-
-    buildPhase = ''
-      buildFlagsArray=(-v -p $NIX_BUILD_CORES -ldflags="-s -w")
-      runHook preBuild
-      go build "''${buildFlagsArray[@]}" -o v2ray ./main
-      go build "''${buildFlagsArray[@]}" -o v2ctl -tags confonly ./infra/control/main
-      runHook postBuild
-    '';
+  nativeBuildInputs = [ makeWrapper ];
 
-    installPhase = ''
-      install -Dm755 v2ray v2ctl -t $out/bin
-    '';
+  installPhase = ''
+    runHook preInstall
+    install -Dm555 "$NIX_BUILD_TOP"/go/bin/main $out/bin/v2ray
+    install -Dm444 release/config/systemd/system/v2ray{,@}.service -t $out/lib/systemd/system
+    install -Dm444 release/config/*.json -t $out/etc/v2ray
+    runHook postInstall
+  '';
 
-    meta = {
-      homepage = "https://www.v2fly.org/en_US/";
-      description = "A platform for building proxies to bypass network restrictions";
-      license = with lib.licenses; [ mit ];
-      maintainers = with lib.maintainers; [ servalcatty ];
-    };
+  assetsDrv = symlinkJoin {
+    name = "v2ray-assets";
+    paths = assets;
   };
 
-in runCommand "v2ray-${version}" {
-  inherit src version;
-  inherit (core) meta;
-
-  nativeBuildInputs = [ makeWrapper ];
+  postFixup = ''
+    wrapProgram $out/bin/v2ray \
+      --set-default V2RAY_LOCATION_ASSET $assetsDrv/share/v2ray
```

Comment:
v2ray now searches XDG_DATA_DIRS for assets.
`--suffix XDG_DATA_DIRS $assetsDrv/share`